src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -967,6 +967,10 @@ protected function incrementOrDecrement($column, $amount, $extra, $method)
             return false;
         }
 
+        if ($this->isClassDeviable($column)) {
+            $amount = (clone $this)->setAttribute($column, $amount)->getAttributeFromArray($column);
+        }
```

Comment:
I'm sorry, but can you explain what this actually does 😅 Why would we clone the model and set the attribute to the amount you want to increment / decrement by?

Reply:
The amount you want to increment / decrement by can be an object, but we need the raw database value (integer, float, ...) in order to perform the increment/decrement: https://github.com/laravel/framework/blob/fc9ee616c5abe690ece735471ac4790aedb2cf7d/src/Illuminate/Database/Eloquent/Model.php#L970

So by setting the attribute on a cloned model using that value (= object) you want to increment / decrement by, we can get the raw database value (integer, float, ...) of that attribute from that cloned model.